cmd/sedad/cmd/config.go

```diff
@@ -81,3 +79,44 @@ func initTendermintConfig() *tmcfg.Config {
 
 	return cfg
 }
+
+func preUpgradeCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "pre-upgrade",
+		Short: "Pre-upgrade command",
+		Long:  "Pre-upgrade command to migrate app.toml for v1.0.0 upgrade",
+		Run: func(cmd *cobra.Command, _ []string) {
+			serverCtx := server.GetServerContextFromCmd(cmd)
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			err := migrateAppConfig(serverCtx, clientCtx.HomeDir)
+			if err != nil {
+				os.Exit(30)
+			}
+			os.Exit(0)
+		},
+	}
+
+	return cmd
+}
+
+func migrateAppConfig(serverCtx *server.Context, rootDir string) error {
+	configPath := filepath.Join(rootDir, "config")
+	appConfigPath := filepath.Join(configPath, "app.toml")
+
+	serverconfig.SetConfigTemplate(serverconfig.DefaultConfigTemplate)
+	oldConfig := serverconfig.DefaultConfig()
+	err := serverCtx.Viper.Unmarshal(oldConfig)
+	if err != nil {
+		return fmt.Errorf("failed to parse %s: %w", appConfigPath, err)
+	}
+
+	defaultSedaConfig := utils.DefaultSEDAConfig()
+	defaultSedaConfig.AllowUnencryptedSEDAKeys = true
```

Comment:
I suppose we're setting this to true here to a node doesn't immediately panic upon updating as it's unlikely operators will have set up the encryption key beforehand? Maybe it's good to add brief comment to explain we set it here rather than modifying the default?

Reply:
Added the comment. We now also require the values of `--key-file-no-encryption` flag and `seda.allow-unencrypted-seda-keys` app config to match when creating a key file.